### PR TITLE
Image block: Caption toolbar is not dismissed when cropping image in place 

### LINKED ISSUE #30334: Gallery block refactor: Caption toolbar is not dismissed when cropping image in place
## Description

Within the new refactored gallery, choose to crop an image in place and accidentally click into the image's caption field, you cannot dismiss the `RichText` toolbar without cancelling image crop mode. 

Report: https://make.wordpress.org/core/2021/03/23/gallery-block-update-call-for-testing/#comment-40970

## Step-by-step reproduction instructions

1. Select image in gallery
2. Click crop button in toolbar
3. Click in image's caption field
4. Attempt to dismiss the caption toolbar by clicking elsewhere in the image or interacting with the toolbar.

## Expected behaviour
The caption toolbar would be dismissed when the caption field blurs.

## Actual behaviour
The caption toolbar persists and interferes with image cropping.

## Screenshots or screen recording (optional)

Video: https://d.pr/v/2ycuq4/BE5iX7xWBa

## WordPress information
- WordPress version: 
- Gutenberg version: 
- Are all plugins except Gutenberg deactivated? 
- Are you using a default theme (e.g. Twenty Twenty-One)? 

## Device information
- Device: 
- Operating system: 
- Browser:

### PR DESCRIPTION
**Pull request by ramonjd**
## Description

When editing an image caption, clicking on the image itself dismisses the caption toolbar. This is normal and expected.

However, when we click on the image's caption field while _cropping an image in place_, we can't dismiss the toolbar without cancelling image crop mode. 

Furthermore the caption field refuses to yield focus. 🤷 

This PR passes an `onBlur` handler down to the RichText package, and blurs the caption field when the field loses focus. 

This needs to be tested in conjunction with https://github.com/WordPress/gutenberg/pull/30587 as the RichText package doesn't currently handle an onBlur callback

We're also blurring away from the caption field when the caption field loses focus.

Resolves https://github.com/WordPress/gutenberg/issues/30334

Also remarked upon in https://github.com/WordPress/gutenberg/issues/23350#issuecomment-647741685

## How has this been tested?
Manually on desktop and a mobile emulator.

#### Testing

1. Check out this PR and also https://github.com/WordPress/gutenberg/pull/30587
2. Select an image in the editor
3. Click the crop button in the toolbar
4. Check that the caption toolbar is dismissed

The caption toolbar should now disappear and the caption field should lose focus. 

#### Before 
Mar-31-2021 22-17-26

#### After
Mar-31-2021 22-13-47

## Types of changes
This is a minor bugfix for an existing block (image).

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/image/image-editing/cropper.js
Comment (ramonjd) [inline packages/block-library/src/image/image-editing/cropper.js]: I'm not entirely sure how accessible this is via keyboard anyway.

Are folks happy with adding a 
```
// eslint-disable-next-line jsx-a11y/no-static-element-interactions

```

to disable for now?

Comment (aaronrobertshaw) [reply packages/block-library/src/image/image-editing/cropper.js]: I don't have any objections to disabling the `no-static-element-interactions` linting for this. It appears to be done in a dozen or more places already.

The accessibility concerns here aren't my strong suit so happy to defer to others there.

**Review by aaronrobertshaw — APPROVED** [review]
This is testing well for me 👍 

The simplicity of being able to click on the image to blur the caption field feels intuitive. Even if the caption field retains focus when selecting options from the cropping toolbar, it is simple to click the image if the caption toolbar is in the way.

I'm happy to give this a tentative approval now but will be good to get some extra eyes on it all the same 🙂

**Comment by glendaviesnz** [comment]
@ramonjd I wonder if we should look for a solution at the richtext toolbar level as the same issue occurs with gallery caption, and I am picking anywhere this component is used?

caption-toolbar-dismiss

**Comment by glendaviesnz** [comment]
Looks like RichText focus management is a longstanding issue - https://github.com/WordPress/gutenberg/issues/9740

**Comment by ramonjd** [comment]
> I wonder if we should look for a solution at the richtext toolbar level as the same issue occurs with gallery caption, and I am picking anywhere this component is used?

Thanks for looking more closely at this @glendaviesnz 

I agree it makes sense to try to address it at the higher level and start a discussion on the best way to approach it.

File: packages/block-library/src/image/image.js
Comment (ramonjd) [inline packages/block-library/src/image/image.js:545]: I don't think we need this ref.blur if https://github.com/WordPress/gutenberg/pull/30587 lands.

**Comment by ramonjd** [comment]
On hold until we resolve a neat way to handle onBlue on RichText components. See: https://github.com/WordPress/gutenberg/pull/30587

**Pull request closed without merging**
